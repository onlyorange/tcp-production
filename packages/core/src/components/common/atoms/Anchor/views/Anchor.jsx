import React from 'react';
import { PropTypes } from 'prop-types';
// eslint-disable-next-line
import Link from 'next/link';
import { buildUrl } from '../../../../../utils/url';
import withStyles from '../../../hoc/withStyles';

import styles from '../Anchor.style';

/**
 * @param {object} props : Props for Anchor
 * @desc This is an anchor component. The prop anchorVariation determines the color of the anchor.
 * The values supported for anchorVariation are : primary, secondary, tertiary.
 * The prop fontSizeVariation determines the font size of the anchor.
 * The value supported for fontSizeVariation are : small, medium, large.
 * The values of these are picked up from the theme.
 * An additional prop noLink is added to support simple anchor or next's Link conditionally.
 */

const Anchor = ({
  children,
  to,
  as,
  className,
  scroll,
  noLink,
  handleLinkClick,
  shallow,
  title,
  target,
  ...other
}) =>
  noLink ? (
    <a
      href={buildUrl(to)}
      className={className}
      {...other}
      onClick={handleLinkClick}
      title={title}
      target={target}
    >
      {children}
    </a>
  ) : (
    <Link href={to} as={as} shallow={shallow} scroll={scroll}>
      <a href={as || buildUrl(to)} className={className} title={title} target={target} {...other}>
        {children}
      </a>
    </Link>
  );

Anchor.propTypes = {
  children: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  as: PropTypes.string,
  scroll: PropTypes.string,
  className: PropTypes.string.isRequired,
  noLink: PropTypes.bool,
  handleLinkClick: PropTypes.func,
  shallow: PropTypes.bool,
  title: PropTypes.string,
  target: PropTypes.string,
};

Anchor.defaultProps = {
  as: '',
  scroll: '',
  noLink: false,
  handleLinkClick: () => {},
  shallow: false,
  title: '',
  target: '',
};

export default withStyles(Anchor, styles);
export { Anchor as AnchorVanilla };
