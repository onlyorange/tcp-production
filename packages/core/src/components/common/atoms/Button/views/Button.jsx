// @flow
import React from 'react';
import type { Node } from 'react';
import withStyles from '../../../hoc/withStyles';

import styles from '../Button.style';

type Props = {
  children: Node,
  className: string,
  ariaLabel: string,
  disabled: boolean,
  fullWidth?: boolean,
  type?: string,
};

/**
 * @param {object} props : Props for button
 * @desc This is a button component. The two variations of buttons are:
 * 1. fixed-width: Takes the width of the column which it occupies.
 * It has the fixed padding as per the zeplin.

 * 2. variable-width: Takes the width of the text that is inside the button.
 * It has fixed padding as per the zeplin. This variation needs to be mentioned in buttonVariation property.
 * TODO - Not able to add these property here due to linting,
 * need to find a way of doing it. Might be resolved with flow types.

 * Additional button Prop:
 * fullWidth: Additional property to mention 100% width of the button.
 * disabled: to have disabled state of the button
 */
const Button = ({ children, className, ariaLabel, disabled, fullWidth, type }: Props): Node => (
  <button
    disabled={disabled}
    aria-label={ariaLabel}
    className={className}
    type={type}
    fullWidth={fullWidth}
  >
    {children}
  </button>
);

Button.defaultProps = {
  fullWidth: true,
  type: 'button',
};

export default withStyles(Button, styles);
export { Button as ButtonVanilla };
