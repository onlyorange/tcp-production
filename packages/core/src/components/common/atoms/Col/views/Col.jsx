// Basic file for column in the grid structure
import React from 'react';
import { PropTypes } from 'prop-types';
import styles from '../Col.style';
import withStyles from '../../../hoc/withStyles';

// Passing on the colConfig to the style File and also the flag to add inline-block to the column
const Col = ({ children, className }) => <div className={className}>{children}</div>;

Col.propTypes = {
  children: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  colSize: PropTypes.shape({
    small: PropTypes.number,
    medium: PropTypes.number,
    large: PropTypes.number,
  }),
};

Col.defaultProps = {
  colSize: {
    small: 6,
    medium: 8,
    large: 12,
  },
};

export default withStyles(Col, styles);
export { Col as ColVanilla };
