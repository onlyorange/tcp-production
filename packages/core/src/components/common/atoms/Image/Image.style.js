import { css } from 'styled-components';

const ImageStyles = css`
  ${props => (props.inheritedStyles ? props.inheritedStyles : '')};
`;

export default ImageStyles;
