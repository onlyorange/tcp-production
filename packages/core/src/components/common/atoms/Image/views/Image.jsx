import React from 'react';
import { PropTypes } from 'prop-types';
import withStyles from '../../../hoc/withStyles';

import styles from '../Image.style';

const Image = ({
  className,
  src,
  srcset,
  sizes,
  placeholderSrc,
  alt,
  inheritedStyles,
  ref,
  ...other
}) => (
  <img
    className={className}
    src={src}
    srcSet={srcset || null}
    sizes={sizes || null}
    alt={alt}
    ref={ref}
    {...other}
    onError={
      placeholderSrc
        ? event => {
            // eslint-disable-next-line no-param-reassign
            event.target.src = placeholderSrc;
          }
        : null
    }
  />
);

Image.propTypes = {
  className: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
  srcset: PropTypes.string,
  sizes: PropTypes.string,
  placeholderSrc: PropTypes.string,
  alt: PropTypes.string.isRequired,
  inheritedStyles: PropTypes.string,
  ref: PropTypes.func,
};

Image.defaultProps = {
  srcset: '',
  sizes: '',
  placeholderSrc: '',
  inheritedStyles: '',
  ref: () => {},
};

export default withStyles(Image, styles);
export { Image as ImageVanilla };
