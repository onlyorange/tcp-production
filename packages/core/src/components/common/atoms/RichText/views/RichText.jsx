import React from 'react';
import { PropTypes } from 'prop-types';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';

import styles from '../RichText.style';

// This accepts an HTML and inserts it wherever the component is added.
const RichText = ({ richTextHtml }) => (
  // eslint-disable-next-line
  <div dangerouslySetInnerHTML={{ __html: richTextHtml }} />
);

RichText.propTypes = {
  richTextHtml: PropTypes.string.isRequired,
};

export default withStyles(RichText, styles);
export { RichText as RichTextVanilla };
