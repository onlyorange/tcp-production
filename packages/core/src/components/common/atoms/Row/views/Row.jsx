import React from 'react';
import { PropTypes } from 'prop-types';
import withStyles from '../../../hoc/withStyles';
import StyledRow from '../Row.style';

// An additonal prop 'fullBleed' is added.
// This property ignores the offset of the row and spans across the space of the grid.
const Row = ({ className, children }) => <div className={className}>{children}</div>;

Row.propTypes = {
  children: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};

export default withStyles(Row, StyledRow);
export { Row as RowVanilla };
