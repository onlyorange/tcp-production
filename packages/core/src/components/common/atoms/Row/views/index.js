// @flow
export { default } from './Row';
