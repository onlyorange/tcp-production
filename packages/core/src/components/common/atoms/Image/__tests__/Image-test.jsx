import React from 'react';
import { shallow } from 'enzyme';
import { ImageVanilla } from '../views/Image';

describe('Image component', () => {
  it('renders correctly', () => {
    const props = {
      className: 'sample-image',
      src: 'acdde.html',
      alt: 'abcd',
    };
    const component = shallow(<ImageVanilla {...props} />);
    expect(component).toMatchSnapshot();
    expect(component.find('.sample-image')).toHaveLength(1);
  });
});
