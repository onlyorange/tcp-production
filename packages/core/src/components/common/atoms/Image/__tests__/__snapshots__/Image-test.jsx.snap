// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component renders correctly 1`] = `ShallowWrapper {}`;
